middleware/common/source/transaction/resource.cpp

```diff
@@ -238,7 +238,7 @@ namespace casual
       {
          auto result = functor();
 
-         if( result != code::xa::resource_fail)
+         if( algorithm::none_of( std::vector{ code::xa::resource_fail, code::xa::resource_error}, predicate::value::equal( result)))
```

Comment:
I don't like that we create a vector each time we check. Perhaps:

```cpp
if( ! algorithm::compare::any( result, code::xa::resource_fail, code::xa::resource_error))
     return result;
     
 ```
 
 Or don't negate it and do the reopen part within the if block.

Reply:
Maybe we want a `algorithm::compare::none( ...)`, it might be more intuitive?